.github/workflows/ci.yml

```diff
@@ -0,0 +1,96 @@
+name: GitHub CI
+
+on:
+  push:
+    paths-ignore:
+      - '.github/*TEMPLATE/**'
+  pull_request:
+    paths-ignore:
+      - '*.md'
+      - '.github/*'
+      - '.github/*TEMPLATE/**'
+
+jobs:
+  build-x64:
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [windows-latest, ubuntu-latest, macos-latest]
+        configuration: [Debug, Release]
+        include:
```

Comment:
Why not add the x86 job here instead of making it a separate one?

Reply:
I described the issues in previous discussion: https://github.com/Cxbx-Reloaded/XbSymbolDatabase/pull/111#discussion_r484204605

Include section does not allow any array nor will read duplicate os string name check. It causes to skip whatever is ordered first and only read the last one for same os name. In order to have same os string is to put it into a separate job section.

Reply:
What if you made the include for `cmake-generator` instead of `os`? (The ugly name could be fixed by using `name`)

Reply:
Is these lines in the link is what you're suggesting?
https://github.com/RadWolfie/XbSymbolDatabaseTestStuff/blob/4fe470701c2f1cc8bddc5d085364a5aa9cfbcf08/.github/workflows/ci.yml#L14-L54

Reply:
Yes

Reply:
Okay, I changed the workflow to the new solution.

Reply:
`plat` can be replaced with `runner.os`

Reply:
Can we rename runner.os to another name such as Android or Xbox One? 😉 I don't think GitHub Actions will have such feature. I do know CMake do have ability compile for other platforms. How? I'm not sure since their generator documentation doesn't say. It's likely necessary to use `-D` argument.

I think I may have another solution for this. By continue use `plat`, except make default string to `runner.os` value. Then override it base on generator string.

Reply:
> 
> 
> Can we rename runner.os to another name such as Android or Xbox One? 😉 I don't think GitHub Actions will have such feature. I do know CMake do have ability compile for other platforms. How? I'm not sure since their generator documentation doesn't say. It's likely necessary to use `-D` argument.
> 
> I think I may have another solution for this. By continue use `plat`, except make default string to `runner.os` value. Then override it base on generator string.

You can continue using `plat` if other things are planned to be supported, then.

Reply:
After thinking about the expression method GitHub provide, I don't think it is possible since expression already resolve into a string from the key's variable. Expression within expression wouldn't work either. Which will also cause issue.

In summary, using `runner.os` wouldn't work for default value since runner haven't run yet which make sense. `defaults` currently only support `run` at the moment.